### DISCUSSION TITLE
v0.4.15

### DISCUSSION THREAD

**Discussion by lbbrhzn** [lbbrhzn/ocpp#352] [Announcements]
## Changes

* add connector status and error code (#351) @lbbrhzn
* Make websocket timeout configurable (#350) @lbbrhzn
* add codecov action (#349) @lbbrhzn

## What's Changed
* add codecov action by @lbbrhzn in https://github.com/lbbrhzn/ocpp/pull/349
* Make websocket timeout configurable by @lbbrhzn in https://github.com/lbbrhzn/ocpp/pull/350
* add connector status and error code by @lbbrhzn in https://github.com/lbbrhzn/ocpp/pull/351

**Full Changelog**: https://github.com/lbbrhzn/ocpp/compare/v0.4.14...v0.4.15

<hr /><em>This discussion was created from the release <a href='https://github.com/lbbrhzn/ocpp/releases/tag/v0.4.15'>v0.4.15</a>.</em>

**Comment by McFranky** [comment]
Hello


ABB charger with Firmware 1.5.2


I got a problem in v0.4.15 with the charge_control switch. In this version the switch dont't show the Charging-Status. It is always off. I can switch the charger on but I don't have the chance to switch it off!

Greeting
Frank

**Reply by lbbrhzn** [reply]
Thanks for reporting. This issue occurs because we now have two statuses, one for the charger and one for the connector. The charging switch is using the wrong status. We'll get it fixed!

**Comment by GrantKeymer** [comment]
I want to say a big **THANKS** for making the Ping Interval and Timeouts configurable.  After playing around with the default values, I found that using a 5 second Ping Interval works much better than the original 20 seconds.  Here is a chart showing before and after...

![HA - Pong Latency 4](https://user-images.githubusercontent.com/35156198/152337830-95e4d56f-386f-45fb-8018-31a9ccb4dcf1.png)

It seems as though the Pulsar Plus has a lazy comms handler which goes into a low power mode if it doesn't see any traffic for a while.  Hitting it with a Ping every 5 seconds seems to keep it running at a better speed, as you can see from the chart above.  Occasional spikes to almost 15 seconds are still there, but the average as seen from the chart has reduced from around 5 seconds to more like 2.  In reality, there are many instances where the ping sits just above 200ms, but the chart above seems to emphasise times well above that.  I've set both timeouts to 20 seconds i.e. well above the expected 5 second interval between pings.

Frequent reconnects are still occurring, mostly between 6 and 20 minutes apart.  Even when I shut down almost everything on my network except my laptop, the RaspPi, the Nokia outdoor LTE router, an access point nearest the Pulsar Plus, and my main router with the DHCP server, reconnects were still happening at much the same rate.  I have no explanation apart from suspecting the Pulsar Plus of having flakey comms.  It has never been terribly robust, especially when using Bluetooth.  Before putting an extra access point in the ceiling above the Pulsar Plus, it used to lose WiFi connection regularly.  That doesn't happen now, so I guess I'll have to live with the reconnects.  Apart from producing a messy chart in HA, they don't seem to matter.

Now I just have one question...

How can I make the Default Ping Interval 5 seconds without needing to edit Const.py each time?
